### ISSUE TITLE
missing resource_details in radical-pilot-stat session json file

### ISSUE THREAD

**Issue by AymenFJA** [radical-cybertools/radical.pilot#2534]
This issue is related to #2496  and #2401. 

How and why:

1. Assuming that the user is using `radical-pilot-stats -m stats -s rp.session` to fetch session statistics.  This command leads to `key errors` for missing resource-related keys like `cores_per_nodes` or `node_list` under the `resource_details` session etc. 
2. Digging into the root issue tells that when https://github.com/radical-cybertools/radical.pilot/blob/75e6e5b0619507d8eaad66566f77e8c67f0944e3/src/radical/pilot/utils/db_utils.py#L58 is pulling the session data from `MongoDB` it returns the pilot information with `resource_details`: `null`, although inside the `rp.session/rp.session.json` the `[pilot][resource_details]` does exist.
3. Example for the json files from inside the session folder vs the one pulled by `radical-pilot-stats`:

downloaded Json  file (when closing/finishing  the session) `rp.session.xxx/rp.session.xxx.json`
```json
   207             "pilot_sandbox": "file://localhost/home/aymen/radical.pilot.sandbox/rp.session.radical.3.aymen.019054.0003/pilot.0000/",
   208             "pmgr": "pmgr.0000",
   209             "resource": "local.localhost",
   210             "resource_details": {
   211                 "rm_info": {
   212                     "agent_node_list": [],
   213                     "cores_per_node": 8,
   214                     "gpus_per_node": 1,
   215                     "lfs_path": "/tmp",
   216                     "lfs_per_node": 1024,
   217                     "mem_per_gpu": null,
   218                     "mem_per_node": 4096,
   219                     "node_list": [
   220                         {
   221                             "cores": [
   222                                 0,
   223                                 0,
   224                                 0,
   225                                 0,
   226                                 0,
   227                                 0,
   228                                 0,
   229                                 0
   230                             ],
   231                             "gpus": [
   232                                 0
   233                             ],
   234                             "lfs": 1024,
   235                             "mem": 4096,

```
vs fetched rp.session.xxx.json via `radical-pilot-stats`

```json

   209             "pilot_sandbox": "file://localhost/home/aymen/radical.pilot.sandbox/rp.session.radical.3.aymen.019054.0003/pilot.0000/",
   210             "pmgr": "pmgr.0000",
   211             "resource": "local.localhost",
   212             "resource_details": null,
   213             "resource_sandbox": "file://localhost/home/aymen/radical.pilot.sandbox",
   214             "resources": {
   215                 "cpu": 8,
   216                 "gpu": 0
   217             },

```

**Comment by andre-merzky** [comment]
I get a different error though:
```sh
$ ./bin/radical-pilot-stats -m stat -s rp.session.rivendell.merzky.019055.0000
modes   : stat
db url  : mongodb://localhost/am
cachedir: /mnt/home/merzky/radical/radical.pilot.devel
session : rp.session.rivendell.merzky.019055.0000
Traceback (most recent call last):
  File "./bin/radical-pilot-stats", line 1100, in <module>
    elif m == 'stat' : stat_session  (db, session, cachedir)
  File "./bin/radical-pilot-stats", line 383, in stat_session
    events     = rpu.get_session_events   (db, session, cachedir=cachedir)
  File "/mnt/home/merzky/radical/radical.pilot.devel/ve3/lib/python3.8/site-packages/radical/pilot/utils/db_utils.py", line 236, in get_session_events
    for event in doc['state_history']:
KeyError: 'state_history'
```

and indeed the `state_history` is not   supported anymore.  I thought so that this is an error you already addressed at some point - am I missing an update maybe?  I am on:
```
$ rs

  python               : /mnt/home/merzky/radical/radical.pilot.devel/ve3/bin/python3
  pythonpath           : 
  version              : 3.8.0
  virtualenv           : /mnt/home/merzky/radical/radical.pilot.devel/ve3

  radical.analytics    : 1.6.7
  radical.gtod         : 1.6.7
  radical.pilot        : 1.12.0-v1.12.0-13-g75e6e5b06@fix-issue_2401
  radical.saga         : 1.12.0
  radical.utils        : 1.12.0
```

**Comment by AymenFJA** [comment]
@andre-merzky: sorry for that, please add `if 'state_history' in doc:` before `for event in doc['state_history']:` in here (as we might not have `state_history` sometimes): https://github.com/radical-cybertools/radical.pilot/blob/75e6e5b0619507d8eaad66566f77e8c67f0944e3/src/radical/pilot/utils/db_utils.py#L236 and here: https://github.com/radical-cybertools/radical.pilot/blob/75e6e5b0619507d8eaad66566f77e8c67f0944e3/src/radical/pilot/utils/db_utils.py#L261

**Comment by andre-merzky** [comment]
Can you please submit a patch? :-)

**Comment by AymenFJA** [comment]
@andre-merzky  I just did.

**Comment by AymenFJA** [comment]
Keys that need to be updated or removed:

```python
   start    = task_doc['statehistory'][0]['timestamp']
```

**Comment by AymenFJA** [comment]
Some progress is made in this PR #2496, this needs a review.

**Comment by AymenFJA** [comment]
This is fixed now in the latest commit to this PR #2490, we now read the already existing `session.xx.json`  instead of pulling the same information from MongoDB.
